backend/prompt_studio/prompt_studio_core_v2/views.py

```
@@ -637,3 +650,89 @@ def import_project(self, request: Request) -> Response:
+                        deployment_types.add("ETL Pipeline")
+                    elif pipeline_type == "TASK":
+                        deployment_types.add("Task Pipeline")
+
+                # Check for Manual Review
+                workflows_with_manual_review = (
+                    WorkflowEndpoint.objects.filter(
+                        workflow_id__in=workflow_ids,
+                        connection_type=WorkflowEndpoint.ConnectionType.MANUALREVIEW,
+                    )
+                    .values_list("workflow_id", flat=True)
+                    .distinct()
+                )
+
+                if workflows_with_manual_review:
+                    deployment_types.add("Human Quality Review")
+
+                deployment_info["deployment_types"] = list(deployment_types)
+
+                # Construct the message if deployments exist
+                if deployment_types:
+                    types_list = sorted(deployment_types)
+                    if len(types_list) == 1:
+                        types_text = types_list[0]
+                    elif len(types_list) == 2:
+                        types_text = f"{types_list[0]} or {types_list[1]}"
+                    else:
+                        types_text = ", ".join(types_list[:-1]) + f", or {types_list[-1]}"
+
+                    deployment_info["message"] = (
+                        f"You have made changes to this Prompt Studio project. "
+                        f"This project is used in {types_text}. "
+                        f"Please export it for the changes to take effect in the deployment(s)."
+                    )
+
+            return Response(deployment_info, status=status.HTTP_200_OK)
+
+        except Exception as e:
+            logger.error(f"Error checking deployment usage for tool {pk}: {e}")
+            return Response(
+                {"error": "Failed to check deployment usage"},
+                status=status.HTTP_500_INTERNAL_SERVER_ERROR,
+            )
```

@athul-rs in case of an error, please raise an exception instead